app.py

```diff
@@ -232,57 +233,81 @@ def find_closest_match(ideal_features, playlist_features):
 
-    local_spotipy = get_spotipy()
     audio_features = None
 
-    if local_spotipy is not None and (
-        local_spotipy.current_playback() is None
-        or local_spotipy.current_playback()["progress_ms"] >= UPDATE_SONG_TIME_MS
-    ):
-        # get the songs from the playlist
-        # tracks = localSP.playlist(playlistID, fields="tracks,next")
-        tracks = local_spotipy.playlist(playlist_id)
-        # print('Custom playlist!')
 
-        # print(tracks)
-        track_ids = []
+    # get the songs from the playlist
+    # tracks = local_spotipy.playlist(playlist_id)
 
-        for (song, i) in zip(tracks["tracks"]["items"], range(100)):
-            if i >= 0 and song is not None:
-                track_ids.append(song["track"]["uri"])
+    track_ids = []
 
-        print(track_ids)
-        # get the audio features
-        audio_features = local_spotipy.audio_features(tracks=track_ids)
+    for (song, i) in zip(songs, range(100)):
+        if i >= 0 and song is not None:
+            track_ids.append(song["track"]["uri"])
 
-        # print('got audio features! print here:')
-        # print(audioFeatures)
-        # find the closest match
-        cool_song = find_closest_match(ideal_audio_features, audio_features)
-        # add to queue
-        if cool_song is not None:
-            local_spotipy.add_to_queue(cool_song)
+    # print(track_ids)
+    # get the audio features
+    audio_features = local_spotipy.audio_features(tracks=track_ids)
 
-            # currently_playing() is not None:
-            # DOESN'T RETURN BOOL, BUT NOONE WILL TELL ME WHAT IT DOES RETURN AND I CANT TEST YET
-            if local_spotipy.current_playback() is not None:
-                local_spotipy.next_track()
-            else:
-                local_spotipy.start_playback()
+    # print('got audio features! print here:')
+    # print(audioFeatures)
+    # find the closest match
+    return find_closest_match(ideal_audio_features, audio_features)
+
+def queue_song(cool_song, spotipy_manager=None):
+    """ takes the song given and queues and plays the song """
+    local_spotipy = get_spotipy() if spotipy_manager is None else spotipy_manager
+
+    # add to queue
+    if cool_song is not None:
+        local_spotipy.add_to_queue(cool_song)
+
+        # currently_playing() is not None:
+        # DOESN'T RETURN BOOL, BUT NOONE WILL TELL ME WHAT IT DOES RETURN AND I CANT TEST YET
+        if local_spotipy.current_playback() is not None:
+            local_spotipy.next_track()
+        else:
+            local_spotipy.start_playback()
+
+# Method that starts play
+def next_song(ideal_audio_features, data):
+    """queues a song based on the ideal audio features"""
+    playlist_id = data["playlistID"]
+    local_spotipy = get_spotipy()
+
+
+    if local_spotipy is not None and (
+            local_spotipy.current_playback() is None
+            or local_spotipy.current_playback()["progress_ms"] >= UPDATE_SONG_TIME_MS
```

Comment:
Can we leave a #TODO note here to remove the song update timing when we get the proper conversation tracking in place